agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiComponentInstaller.java

```diff
@@ -128,9 +128,19 @@ private void start(Instrumentation instrumentation) {
     }
 
     File javaTmpDir = new File(System.getProperty("java.io.tmpdir"));
+    boolean readOnlyFileSystem = false;
+    if (javaTmpDir.canRead() && !javaTmpDir.canWrite()) {
+      readOnlyFileSystem = true;
+    }
+
     File tmpDir = new File(javaTmpDir, "applicationinsights-java");
-    if (!tmpDir.exists() && !tmpDir.mkdirs()) {
-      throw new IllegalStateException("Could not create directory: " + tmpDir.getAbsolutePath());
+    if (!readOnlyFileSystem) {
+      if (!tmpDir.exists() && !tmpDir.mkdirs()) {
+        throw new IllegalStateException("Could not create directory: " + tmpDir.getAbsolutePath());
+      }
+    } else {
+      startupLogger.info(
+          "Could not create directory in a read only file system: " + tmpDir.getAbsolutePath());
```

Comment:
this doesn't help the user understand the implications. I think better to log a message about telemetry not being stored to disk on sporadic network failures. And also check here to see if profiler is enabled, and log message about that not being supported with read-only file system.

Reply:
don't log anything here then.. log at each case.. e.g. disk persistence is not supported with read-only file system, profiler is not supported with a read-only file system?

Reply:
message has been updated. see if it's better now.